### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by wilmouths**
* Update code
* Lint code
* Update dependancies

### CODE REVIEW

File: app/controllers/metrics/metrics.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const mongo = require('../../lib/mongo');
+const cache = require('../../lib/cache');
+const logger = require('../../lib/logger');
+
+function getBadgeInDatabase(badgeId) {
```
Comment (nojhamster) [inline app/controllers/metrics/metrics.js:5]: Something like `getBadgeCount()` would be more suitable, as this function does not fetch the actual badge.

File: app/controllers/badges/badges.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    const { badges } = cache.get();
+
+    if (!result || !result.badges) {
+      return res.json({ status: 'success', data: { badges } });
+    }
+
+    const userBadges = [];
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const badge of badges) {
```
Comment (nojhamster) [inline app/controllers/badges/badges.js:26]: A `map()` would be more idiomatic than pushing in a loop, and wouldn't require to disable `no-restricted-syntax`

File: app/controllers/badges/badges.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+      badges: userBadges,
+      visibility: !!result.visibility,
+    };
+
+    return res.json({ status: 'success', data });
+  });
+};
+
+function getBagdeInfos(badgeId) {
+  return new Promise(async (resolve, reject) => {
```
Comment (nojhamster) [inline app/controllers/badges/badges.js:53]: What's the point of using both Promise and async ?

**Comment by nojhamster** [comment]
Sync methods should be avoided here https://github.com/ezpaarse-project/ezpaarse-badge/blob/master/app/lib/api.js#L13-L16

File: app/controllers/share/share.js [hunk 4]
```diff
@@ -76,44 +62,49 @@ async function render(req, res, view) {
...
 
   if (!user) {
-    error.message = 'noUserFound';
-    return res.status(404).render('error', error);
+    return errorHandler(req, res, { message: 'noUserFound' });
   }
-  user = JSON.parse(user);
+
+  const style = {
+    css: fs.readFileSync(path.resolve('public', 'css', `${view}.css`), 'utf-8'),
```
[the remark this answers is not in the document]
Comment (wilmouths) [reply app/controllers/share/share.js:78]: Added fs-extra to use promises

**Pull request merged**